### DISCUSSION TITLE
loading TARDIS image plates

### DISCUSSION THREAD

**Discussion by amylazicki** [HEXRD/hexrdgui#1399] [Q&A] [answered]
I am running hexrdgui on a PC, using the most recent stable release.  I encounter a few buggy things on loading the image plates.  When I load the plate, I have to add the detector outline in order to remember which transform I need to apply to get it oriented properly, but then when I apply the transform, the detector outline disappears and the button to Add Dectector Outline is greyed out.  I have to push 'ok' under 'Position Detector Outline' to get the button back.  Is there a particular reason for enforcing a certain 'order of operations' in this panel?  Can't all the buttons be kept active?  Another funny thing is that when I move the image plate outline, the original position sticks around as a dashed line:

![image](https://user-images.githubusercontent.com/130735505/232550453-373a78cf-cf96-4d4b-9144-9f0051ba4758.png)

which is ok except when the original position is actually very close to the correct location, and then the dashed line makes it harder to see the edges of hte image plate, and whether I have moved the outline to the right place.

**Comment by bnmajor** [comment, marked as answer]
Hi @amylazicki!

> When I load the plate, I have to add the detector outline in order to remember which transform I need to apply to get it oriented properly, but then when I apply the transform, the detector outline disappears and the button to Add Dectector Outline is greyed out.

Which transform are you applying when you see this behavior? I have been unable to reproduce on master or `v0.9.2` so far. I know that the last release we had (0.9.2) had a bug where a line would be drawn on the TARDIS plates but it seems like I am still able to apply a transform and keep the template:
![tardis_transform_0 9 2](https://user-images.githubusercontent.com/51238406/232570795-74cd01de-63ae-4d39-afc0-1f46ceb21f3a.gif)

On master the bug that produced that line is fixed:
![tardis_transform_master](https://user-images.githubusercontent.com/51238406/232570900-69a6fec0-85a8-464d-9b81-a1c2af86ec5f.gif)

>  I have to push 'ok' under 'Position Detector Outline' to get the button back.

If the detector outline is dissapearing that certainly is a bug that needs to be addressed! Hopefully if you share the steps you are taking I will be able to reproduce and fix that. Worth noting though: if you have just changed your mind/made a mistake/want to re-do the current image plate you can instead select `Cancel` under the`Postion Detector Outline` section.

> Is there a particular reason for enforcing a certain 'order of operations' in this panel? Can't all the buttons be kept active?

The reason that there is a certain flow enforced for things is because it can be very difficult to make assumptions about the users intentions if things are done out of order. However, if this "order of operations" enforcement is more problematic than helpful this approach can be revisited! I am currently working on a handful of changes for this tool (see below) and I will think about how this can be better handled while I work on those other improvements.

> Another funny thing is that when I move the image plate outline, the original position sticks around as a dashed line

I will make an issue for this! In my testing it seems the last applied template is what is used and it is just the canvas that is not being cleaned up properly.

There have been a lot of good suggestions for improving the LLNL Import Tool recently so to summarize some in-progress improvements:

1. Adding support for loading `.gel` images
2. Applying a default transform for any/all LLNL image plates that always (or very often) need the same transform
3. Automatically applying the detector outline once the image plate and image have been selected (so far only for TARDIS since PXRDIP and BBXRD (coming soon) use a single image for all plates)
4. Simplifying the flow and using more explicit wording to address some of the issues that you were seeing

I will also let you know when the documentation for the import tool has been updated and hopefully that will also help!

**Reply by amylazicki** [reply]
This is all working well for me now!  Thank you!
